src/SymArrayDmap.chpl

```diff
@@ -79,15 +79,19 @@ module SymArrayDmap {
     proc makeDistArray(in a: [?D] ?etype) throws
       where D.rank == 1 && (MyDmap == Dmap.defaultRectangular || !a.isDefaultRectangular())
     {
-      var res = D.tryCreateArray(etype);
+      var res = makeDistDom((...D.shape)).tryCreateArray(etype);
       res = a;
       return res;
     }
 
     proc makeDistArray(in a: [?D] ?etype) throws
       where D.rank > 1 && (MyDmap == Dmap.defaultRectangular || !a.isDefaultRectangular())
     {
-      return a;
+      const dom = makeDistDom((...D.shape));
+      var res: [dom] etype;
+      res = a;
+      return res;
```

Comment:
What is the point of these changes? These have had performance impacts in the past, so I think we need to make sure they are OK before changing thee. Noting that the first function has a `where` clause, so we know it's only 1D and shouldn't need to do tuple unrolling there.

Reply:
When reading a Zarr store, the `readZarrArray` method returns a distributed array. If I, without this code change, try to add that array to the symbol table, I get the following error: 
```
/Users/brandonneth/arkouda//src/MultiTypeSymEntry.chpl:261: error: cannot assign expression of type [BlockDom(1,int(64),one,unmanaged DefaultDist)] real(64) to field 'a' of type [domain(1,int(64),one)] real(64)
```

I think this is caused by the fact that the generic `SymEntry` class only supports the `a` field having the default type of `makeDistArray`. I ended up reverting the code in this file and instead creating an array with the correct type and copying into it within the ZarrMsg.chpl file.